crates/holochain/src/conductor/space.rs

```diff
@@ -297,9 +318,133 @@ impl Spaces {
         Ok(results)
     }
 
-    #[instrument(skip(self, op_hashes))]
+    /// The network module needs info about various groupings ("regions") of ops
+    ///
+    /// Note that this always includes all ops regardless of integration status.
+    /// This is to avoid the degenerate case of freshly joining a network, and
+    /// having several new peers gossiping with you at once about the same regions.
+    /// If we calculate our region hash only by integrated ops, we will experience
+    /// mismatches for a large number of ops repeatedly until we have integrated
+    /// those ops. Note that when *sending* ops we filter out ops in limbo.
```

Comment:
I'd have to think about this some more, but I feel like this attempt at optimizing one case may be making other scenarios worse. E.g. if there are bad actors or bugs out there producing ops that won't integrate, or have dependencies to things that cannot be fetched, won't this make everybody's regions mismatch all the time?

Reply:
That's true. In that case though it will only be the bad actor and their gossip partners who will experience mismatches, and everyone else will gossip just fine.

Maybe we should send two separate hashes for each region, integrated and limbo ops? Potentially could even crunch the 32 byte hashes down to 16 bytes each to keep the payload the same size. Then we have more possibilities for a match.